modules/rostests/apitests/ntdll/NtAllocateVirtualMemory.c

```diff
@@ -559,42 +563,202 @@ CheckAdjacentVADs()
     Size = 0x10000;
     Status = NtFreeVirtualMemory(NtCurrentProcess(), &BaseAddress, &Size, MEM_RELEASE);
     ok_ntstatus(Status, STATUS_SUCCESS);
-
 }
 
-#define RUNS 32
-
-START_TEST(NtAllocateVirtualMemory)
+static
+VOID
+CheckSomeDefaultAddresses(VOID)
 {
-    PVOID Mem1, Mem2;
-    SIZE_T Size1, Size2;
-    ULONG i;
     NTSTATUS Status;
+    PVOID BaseAddress;
+    SIZE_T Size;
 
-    CheckAlignment();
-    CheckAdjacentVADs();
+    // NULL.
+
+    /* Reserve memory dynamically, not at 0x00000000 */
+    BaseAddress = NULL;
+    Size = 0x1000;
+    Status = NtAllocateVirtualMemory(NtCurrentProcess(),
+                                     &BaseAddress,
+                                     0,
+                                     &Size,
+                                     MEM_RESERVE,
+                                     PAGE_READWRITE);
+    ok_ntstatus(Status, STATUS_SUCCESS);
+    ok(BaseAddress != 0x00000000, "Unexpected BaseAddress = 0x00000000\n");
+    Status = NtFreeVirtualMemory(NtCurrentProcess(), &BaseAddress, &Size, MEM_RELEASE);
+    ok_ntstatus(Status, STATUS_SUCCESS);
+
+    // 0x00000000, 64k: Free.
+
+    /* Reserve and commit memory at 0x00000000, after round down */
+    BaseAddress = UlongToPtr(0x00000000 + 0x0FFF);
+    Size = 0x1000;
+    Status = NtAllocateVirtualMemory(NtCurrentProcess(),
+                                     &BaseAddress,
+                                     0,
+                                     &Size,
+                                     MEM_RESERVE | MEM_COMMIT,
+                                     PAGE_READWRITE);
+    ok_ntstatus(Status, STATUS_SUCCESS);
+    ok_ptr(BaseAddress, 0x00000000);
+
+    // Double-check that it is not forbidden "in order to catch null pointer accesses".
+    _SEH2_TRY
+    {
+        *(int*)UlongToPtr(0x00000000) = 1;
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
```

Comment:
There's StartSeh/EndSeh to simplify this

Reply:
Thanks for the hint. Done.